test/modifiers.c

```diff
@@ -151,10 +157,67 @@ test_modmap_none(void)
     xkb_context_unref(context);
 }
 
+static void
+test_real_mods(void)
+{
+    struct xkb_context *context = test_get_context(0);
+    struct xkb_keymap *keymap;
+    const struct xkb_key *key;
+    xkb_keycode_t keycode;
+
+    keymap = test_compile_file(context, "keymaps/real-modifiers.xkb");
+    assert(keymap);
+
+    keycode = xkb_keymap_key_by_name(keymap, "AD01");
+    assert(keycode != XKB_KEYCODE_INVALID);
+    key = XkbKey(keymap, keycode);
+    assert(key->modmap == Mod5Mask);
+    assert(key->vmodmap == Mod10Mask << 1);
+
+    keycode = xkb_keymap_key_by_name(keymap, "AD02");
+    assert(keycode != XKB_KEYCODE_INVALID);
+    key = XkbKey(keymap, keycode);
+    assert(key->modmap == Mod6Mask);
+    assert(key->vmodmap == Mod10Mask << 2);
+
+    assert(test_key_seq(keymap,
+                        KEY_O, BOTH, XKB_KEY_a, NEXT,
+                        KEY_Q, DOWN, XKB_KEY_q, NEXT,
+                        KEY_O, BOTH, XKB_KEY_c, NEXT,
+                        KEY_Q, UP,   XKB_KEY_q, NEXT,
```

Comment:
Is there any specific reason for this whole test sequence, with some keys down/up and others as `BOTH`? It looks special enough but I can't see it. Other than the up/down keys are our new modifiers but we don't actually check for modifier state anywhere.

Also, I must be missing something but: why do `a`, `c`, ... have a keysym assigned? our keymap doesn't use `include` so shouldn't all those be `NoSymbol`?